internal/highway/x/ipfs/protocol.go

```diff
@@ -49,13 +52,32 @@ func DecodeCIDFromString(s string) (cid.Cid, error) {
 }
 
 // GetData returns a file from IPFS.
-func (i *IPFSProtocol) GetData(cid string) ([]byte, error) {
+func (i *Protocol) GetData(cid string) ([]byte, error) {
 	// Decode CID from String
 	c, err := DecodeCIDFromString(cid)
 	if err != nil {
 		return nil, err
 	}
 
+	ctx := context.TODO()
+	key := datastore.NewKey(cid)
+
+	exists, err := i.store.Has(ctx, key)
+	if err != nil {
+		// log error and continue
+		log.Println(err)
```

Comment:
Is it possible for exists to be `true` and err to not be nil?

If not I would suggest returning the error here.